perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2984,120 +3183,919 @@ private void closeExtSourcesAfterSynchronization(ExtSource membersSource, ExtSou
+		String skippedMembersMessage = null;
+		boolean failedDueToException = false;
+		try {
+			List<String> skippedMembers = perunBl.getGroupsManagerBl().synchronizeGroup(sess, group);
+
+
+			skippedMembersMessage = prepareSkippedObjectsMessage(skippedMembers, "members");
+			exceptionMessage = skippedMembersMessage;
+
+		} catch (WrongAttributeValueException | WrongReferenceAttributeValueException | InternalErrorException |
+				WrongAttributeAssignmentException | GroupNotExistsException |
+				AttributeNotExistsException | ExtSourceNotExistsException e) {
+			failedDueToException = true;
+			exceptionMessage = "Cannot synchronize group ";
+			log.error(exceptionMessage + group, e);
+			exceptionMessage += "due to exception: " + e.getName() + " => " + e.getMessage();
+		} catch (Exception e) {
+			failedDueToException = true;
+			exceptionMessage = "Cannot synchronize group ";
+			log.error(exceptionMessage + group, e);
+			exceptionMessage += "due to unexpected exception: " + e.getClass().getName() + " => " + e.getMessage();
+		} finally {
 			try {
-				removeRelationMembers(sess, resultGroup, getGroupMembers(sess, operandGroup), operandGroup.getId());
-			} catch (WrongAttributeValueException | WrongReferenceAttributeValueException ex) {
-				throw new InternalErrorException("Removing relation members failed. Cause: {}", ex);
-			} catch (NotGroupMemberException ex) {
-				throw new ConsistencyErrorException("Database inconsistency. Cause: {}", ex);
+				perunBl.getGroupsManagerBl().saveInformationAboutGroupSynchronizationInNestedTransaction(sess, group, failedDueToException, exceptionMessage);
+			} catch (Exception ex) {
+				log.error("When synchronization group " + group + ", exception was thrown.", ex);
+				log.info("Info about exception from synchronization: " + skippedMembersMessage);
 			}
-		} else {
-			throw new GroupRelationCannotBeRemoved("Union between result group " + resultGroup + " and operand group" + operandGroup +
-					" cannot be removed, because it's part of the hierarchical structure of the groups.");
 		}
+	}
 
-		groupsManagerImpl.removeGroupUnion(sess, resultGroup, operandGroup);
+	/**
+	 * Compose message about skipped objects
+	 *
+	 * Method used by group structure synchronization
+	 *
+	 * @param skippedObjects objects which were skipped during synchronization
+	 * @param objectName name of objects which were skipped
+	 * @return composed message about skipped objects
+	 */
+	private String prepareSkippedObjectsMessage(List<String> skippedObjects, String objectName) {
+		String skippedObjectsMessage = null;
 
-		// recalculates statuses of members in result group
-		for (Member member : getGroupMembers(sess, resultGroup)) {
-			recalculateMemberGroupStatusRecursively(sess, member, resultGroup);
+		if (!skippedObjects.isEmpty()) {
+			skippedObjectsMessage = "These " + objectName + " from extSource were skipped: { ";
+
+			for (String skippedGroup : skippedObjects) {
+				if (skippedGroup == null) continue;
```

Comment:
You are using this for skipped groups and members, but the skipped object here is defined as skippedGroup. It should be skippedObject.